src/main/java/graphql/execution/ExecutionContext.java

```diff
@@ -13,6 +15,7 @@
 
 public class ExecutionContext {
 
+    private final Instrumentation instrumentation;
     private GraphQLSchema graphQLSchema;
```

Comment:
The other variables in this class should be final.  The setters are an anti pattern.